src/lex/cpp.rs

```diff
@@ -772,7 +783,8 @@ impl<'a> PreProcessor<'a> {
                     .push_back(location.with(PendingToken::Replacement(token.clone())));
             }
         }
-        self.next()
+        let first = self.pending.pop_front()?;
```

Comment:
```suggestion
        // NOTE: no errors could have occurred while parsing this function body
        // since they would have returned before getting here.
        let first = self.pending.pop_front()?;
```